bika/lims/browser/auditlog.py

```diff
@@ -0,0 +1,309 @@
+# -*- coding: utf-8 -*-
+
+import collections
+import json
+
+from bika.lims import api
+from bika.lims import bikaMessageFactory as _
+from bika.lims import logger
+from bika.lims.browser.bika_listing import BikaListingView
+from bika.lims.interfaces import IAuditable
+from bika.lims.utils import t
+from plone.memoize import view
+from Products.CMFPlone.i18nl10n import ulocalized_time
+from Products.Five.browser import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+
+
+class HasAuditLog(BrowserView):
+    """Helper View to check if the context provides the IAuditable interface
+    """
+    def __call__(self):
+        return IAuditable.providedBy(self.context)
+
+
+class AuditLogView(BikaListingView):
+    """Audit View
+    """
+    template = ViewPageTemplateFile("templates/audit.pt")
+    diff_template = ViewPageTemplateFile("templates/audit_diff.pt")
+
+    def __init__(self, context, request):
+        super(AuditLogView, self).__init__(context, request)
+
+        self.show_select_column = False
+        self.show_search = False
+        self.pagesize = 5
+
+        self.icon = "{}/{}/{}".format(
+            self.portal_url,
+            "++resource++bika.lims.images",
+            "%s_big.png" % context.portal_type.lower())
+
+        self.title = "{} Log".format(api.get_title(context))
+
+        self.columns = collections.OrderedDict((
+            ("version", {
+                "title": _("Version"), "sortable": False}),
+            ("time", {
+                "title": _("Time"), "sortable": False}),
+            ("modified", {
+                "title": _("Date Modified"), "sortable": False}),
```

Comment:
As far as I understand, `time` displays the exact time the snapshot took place, while `modified` displays the value for the attribute with same name from the object/brain. I think both fields will always have same value (maybe with 1 sec. of difference). I would suggest to display one of the two in the listing (I prefer "modified", but with `_("Time")` as the column title).

Reply:
`time` comes actually from the review history and is there to simply pass in the whole review history when creating the snapshot. Since no modification takes place in this case, the time shows the timestamp of the workflow transition